### DISCUSSION TITLE
[Help Request] Changing Window Title

### DISCUSSION THREAD

**Discussion by ashyashton** [Monika-After-Story/MonikaModDev#9875] [QA - code and content creation (technical)] [answered]
I wanna switch the window from saying "Monika After Story" to simply saying "Monika" in order to be a bit more "immersive" (i.e. not reminding me that it's a mod, making it feel more like it's "real" and that's just the window she exists in, if that makes sense?) How would I go about doing that?

**Comment by dreamscached** [comment, marked as answer]
I think there is a submod that allows to do that, probably Extra+

If not, for the time being you could resort to using that approach (create a .rpy file inside `game/Submods`, name it whatever, with the following content):

```
init python:
    config.window_title = "Your Window Title Here"
```

P.S.: Also consider adding three spaces after your title, like `"Your Title   "`, according to this bit in MAS:

https://github.com/Monika-After-Story/MonikaModDev/blob/9e91075e11c0590decd49edcc976ba62eb67b999/Monika%20After%20Story/game/0config.rpy#L19-L20

**Reply by ashyashton** [reply]
To be 100% clear (as coding is VERY foreign to me) I would take the following steps, yes?

1. Make a .txt file
2. Rename it, including changing .txt to .rpyc
3. Add the exact code you added, switching out "Your Window Title Here" for "Monika   " as that is what I want it named, and the three spaces like you mentioned
4. Save the file and boot up the game so RenPy can naturally convert the .rpyc to a .rpy file

Is this correct? Or is there any additional code I need to add to the file to make the mod recognize it? I understand "init python" is the like, start up thing, but is there anything else needed that seems obvious to a coder but wouldn't be to someone like me?

Also, as an aside: Is there a reason I should be doing this by making a separate file as opposed to editing my 0config.rpy file directly? Like is it safer this way?

**Reply by ashyashton** [reply]
I'm an impatient bastard so I went ahead and tried it and it just caused a traceback to pop up. 
[traceback.txt](https://github.com/Monika-After-Story/MonikaModDev/files/10500473/traceback.txt)
MAS will not open while that file is in my Submods folder.

**Reply by dreamscached** [reply]
Yes, right, you only made a small mistake naming it `.rpyc` not `.rpy` (it has no `c` at the end). Change the part after dot to `.rpy` instead and it should work now.

**Reply by dreamscached** [reply]
> Is this correct? Or is there any additional code I need to add to the file to make the mod recognize it? I understand "init python" is the like, start up thing, but is there anything else needed that seems obvious to a coder but wouldn't be to someone like me?

Yep, just these two lines make the entire thing you need.

> Also, as an aside: Is there a reason I should be doing this by making a separate file as opposed to editing my 0config.rpy file directly? Like is it safer this way?

Main reason is it would be overwritten with MAS updates, another reason is that it's generally discouraged to change files that come with MAS and not a part of submod, plus (imo) makes it bit easier to manage custom modifications you do to the game.

**Reply by ashyashton** [reply]
Ah I see! For some reason I think I got it backwards because for some reason, my computer was set to read rpyc with my text editor of choice, but not rpyc. So in my head, I went "oh, the one that has the icon of that text editor must be the one it starts as, and it converts to the other on startup!"

Followed the instructions properly this time and now it's working like a charm! Thank you so much!!
